src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,2389 @@
+			count_SAT_OC_MetOnce = 0;
+			count_DidWeMeetLoad = 0;
+			count_DidWeNotMeetLoad = 0;
+			// vector below used store the modes in each timestep that don't meet humidity or temperature limits, used in warnings
+			std::vector<int> temp(25);
+			SAT_OC_MetinMode_v= temp;
+			SAHR_OC_MetinMode_v= temp;
+			
+			int MAXIMUM_OPERATIONAL_SETTINGS = 5;
+			ModeCounter = 0;
+			//!!!!!!!!!!!!!!!!!!!!!!!!! this debug code will be removed nearer the code freeze, please don't comment on it, it will be gone.
+			DebugBreak = 12; // remove
+	
+			CurrentOperatingSettings.resize(5);
+
+			InitializeModelParams();
+		}
+		
+		void Model::ResetOutputs()
+		{
+			UnitTotalCoolingRate = 0;
+			UnitTotalCoolingEnergy = 0;
+			UnitSensibleCoolingRate = 0;
+			UnitSensibleCoolingEnergy = 0;
+			UnitLatentCoolingRate = 0;
+			UnitLatentCoolingEnergy = 0;
+			SystemTotalCoolingRate = 0;
+			SystemTotalCoolingEnergy = 0;
+			SystemSensibleCoolingRate = 0;
+			SystemSensibleCoolingEnergy = 0;
+			SystemLatentCoolingRate = 0;
+			SystemLatentCoolingEnergy = 0;
+			UnitTotalHeatingRate = 0;
+			UnitTotalHeatingEnergy = 0;
+			UnitSensibleHeatingRate = 0;
+			UnitSensibleHeatingEnergy = 0;
+			UnitLatentHeatingRate = 0;
+			UnitLatentHeatingEnergy = 0;
+			SystemTotalHeatingRate = 0;
+			SystemTotalHeatingEnergy = 0;
+			SystemSensibleHeatingRate = 0;
+			SystemSensibleHeatingEnergy = 0;
+			SystemLatentHeatingRate = 0;
+			SystemLatentHeatingEnergy = 0;
+			SupplyFanElectricPower = 0;	   
+			SupplyFanElectricEnergy = 0;    
+			SecondaryFuelConsumptionRate = 0;
+			SecondaryFuelConsumption = 0;   
+			ThirdFuelConsumptionRate = 0;   
+			ThirdFuelConsumption = 0;	   
+			WaterConsumptionRate = 0;	   
+			WaterConsumption = 0;		   
+			ExternalStaticPressure = 0;	 
+			QLatentZoneOut = 0;
+			QSensZoneOut = 0;
+			//debug
+			//RawHR = 0;
+		}
+
+		void Model::InitializeModelParams()
+		{
+			// SUBROUTINE INFORMATION:
+			//       AUTHOR         Spencer Maxwell Dutton
+			//       DATE WRITTEN   October 2017 
+			//       MODIFIED       
+			//       RE-ENGINEERED  na
+
+			// PURPOSE OF THIS SUBROUTINE:
+			// Reset calculation values 
+
+			// METHODOLOGY EMPLOYED:
+			// 
+
+			// REFERENCES:
+			// na
+
+			// Using/Aliasing
+			ResetOutputs();
+			PrimaryMode = 0;
+			PrimaryModeRuntimeFraction = 0;
+			optimal_EnvCondMet = false;
+			Tsa = 0;
+			//reset the power use to a high value, this is replaced during the calculation keeping the "best" setting. 
+			//pOptimal.ElectricalPower = IMPLAUSIBLE_POWER;
+			RunningPeakCapacity_EnvCondMet = false;
+			Settings.clear();
+		}
+
+		Model::~Model()                 
+		{
+			for (auto iter = SolutionSpaces.begin(); iter != SolutionSpaces.end(); ++iter) {
+				CModeSolutionSpace* p = (*iter);
+				delete p;
+			} 
+			
+			for (auto iterSettings = Settings.begin(); iterSettings != Settings.end(); ++iterSettings) {
+				 (*iterSettings).reset();
+			}
+		}
+
+		CMode::~CMode()  
+		{
+	
+		}
+
+		int Model::GetID()
```

Comment:
Although getters and setters are a good idea in most projects, we don't actually need them in E+.  Just make the ID variable a public member and grab it directly.  If this function is intended to do more work before returning the member, such as looking up some other data, then sure, a mining function would be OK.

Reply:
ok removed those.